mono/mini/interp/transform.c

```diff
@@ -2861,8 +2871,10 @@ generate (MonoMethod *method, RuntimeMethod *rtm, unsigned char *is_bb_start, Mo
 			++td.ip;
 			break;
 		case CEE_ENDFINALLY:
+			g_assert (td.clause_indexes [in_offset] != -1);
 			td.sp = td.stack;
 			SIMPLE_OP (td, MINT_ENDFINALLY);
+			ADD_CODE (&td, td.clause_indexes [in_offset]);
```

Comment:
you should change the size in `mintops.def` for `MINT_ENDFINALLY` and increase the `ip` properly in the interpreter loop for that instruction.